pkg/scheduler/plugins/capacity/capacity.go

```diff
@@ -165,25 +165,38 @@ func (cp *capacityPlugin) OnSessionOpen(ssn *framework.Session) {
 		return !overused
 	})
 
+	queueAllocatable := func(queue *api.QueueInfo, candidate *api.TaskInfo) bool {
+		attr := cp.queueOpts[queue.UID]
+		futureUsed := attr.allocated.Clone().Add(candidate.Resreq)
+		allocatable := futureUsed.LessEqualWithDimension(attr.realCapability, candidate.Resreq)
+		if !allocatable {
+			klog.V(3).Infof("Queue <%v>: realCapability <%v>, allocated <%v>; Candidate <%v>: resource request <%v>",
+				queue.Name, attr.realCapability, attr.allocated, candidate.Name, candidate.Resreq)
+		}
+		return allocatable
+	}
+
 	ssn.AddAllocatableFn(cp.Name(), func(queue *api.QueueInfo, candidate *api.TaskInfo) bool {
 		if !readyToSchedule {
 			klog.V(3).Infof("Capacity plugin failed to check queue's hierarchical structure!")
 			return false
 		}
-
 		if hierarchyEnabled && !cp.isLeafQueue(queue.UID) {
+			klog.V(3).Infof("Queue <%s> is not a leaf queue, can not allocate task <%s>.", queue.Name, candidate.Name)
 			return false
 		}
-		attr := cp.queueOpts[queue.UID]
-
-		futureUsed := attr.allocated.Clone().Add(candidate.Resreq)
-		allocatable := futureUsed.LessEqualWithDimension(attr.realCapability, candidate.Resreq)
-		if !allocatable {
-			klog.V(3).Infof("Queue <%v>: realCapability <%v>, allocated <%v>; Candidate <%v>: resource request <%v>",
-				queue.Name, attr.realCapability, attr.allocated, candidate.Name, candidate.Resreq)
+		list := append(cp.queueOpts[queue.UID].parents, queue.UID)
```

Comment:
The assembly of list also looks very strange. Why does it become [queue.father, queue.grandfather, ..., queue.ancestor, queue]? Wasn’t the previous writing method quite good? Why was it changed to this?